tests/apps/courses/test_templates_course_detail.py

```diff
@@ -443,6 +447,95 @@ def test_templates_course_detail_no_programs(self):
         self.assertEqual(response.status_code, 200)
         self.assertNotContains(response, "course-detail__programs")
 
+    def test_templates_course_detail_teaser_empty_cover_empty(self):
+        """
+        Without video in teaser and no image in cover placeholders,
+        no component should be present on the teaser placeholder.
+        """
+        user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=user.username, password="password")
+
+        course = CourseFactory()
+
+        response = self.client.get(course.extended_object.get_absolute_url())
+
+        self.assertEqual(response.status_code, 200)
+        pattern = (
+            r'<div class="subheader__teaser">'
+            r'<p class="empty">'
+            r"Add a video on teaser or add an image for course cover on its glimpse."
+            r"</p>"
+            r"</div>"
+        )
+        self.assertIsNotNone(re.search(pattern, str(response.content)))
+
+    def test_templates_course_detail_teaser_video_cover_empty(self):
+        """
+        With video in teaser and no image in cover placeholders,
+        video component should be present on the teaser placeholder.
+        """
+        user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=user.username, password="password")
+
+        video_sample = random.choice(VIDEO_SAMPLE_LINKS)
+        course = CourseFactory(fill_teaser=video_sample)
+
+        response = self.client.get(course.extended_object.get_absolute_url())
+
+        self.assertEqual(response.status_code, 200)
+        pattern = (
+            r'<div class="subheader__teaser">'
+            r'<div class="aspect-ratio">'
+            r'<iframe src="{url:s}"  allowfullscreen></iframe>'
+            r"</div>"
+            r"</div>"
+        ).format(url=video_sample.url)
+        self.assertIsNotNone(re.search(pattern, str(response.content)))
+
+    def test_templates_course_detail_teaser_empty_cover_image(self):
+        """
+        Without video in teaser and with image in cover placeholders,
+        image component should be present on the teaser placeholder.
+        """
+        user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=user.username, password="password")
+
+        video_sample = random.choice(VIDEO_SAMPLE_LINKS)
+        course = CourseFactory(fill_cover=pick_image("cover")(video_sample.image))
```

Comment:
You could publish the page and remove login part.
```suggestion
        course = CourseFactory(fill_cover=pick_image("cover")(video_sample.image), should_publish=True)
```